nixos/modules/services/databases/postgresql.nix

```diff
@@ -123,7 +123,88 @@ in
       };
 
       ensureDatabases = mkOption {
-        type = types.listOf types.str;
+        type = let
+          databaseType = types.submodule {
+            options = {
+              name = mkOption {
+                type = types.str;
+                description = lib.mdDoc ''
+                  Name of the database to ensure.
+                '';
+              };
+
+              owner = mkOption {
+                type = types.nullOr types.str;
+                default = null;
+                example = "db_user";
+                description = lib.mdDoc ''
+                  Name of the user that should own the new database.
+                '';
+              };
+
+              template = mkOption {
+                type = types.nullOr types.str;
+                default = null;
```

Comment:
What uses postgres when creating a DB? We should default to that

Reply:
I suspect that most of these options might have different defaults depending on the postgresql version.

Some of these could also be influenced by options for the daemon itself